packages/@mantine/core/src/components/Switch/Switch.tsx

```diff
@@ -208,12 +210,12 @@ export const Switch = factory<SwitchFactory>((_props, ref) => {
         id={uuid}
         ref={ref}
         type="checkbox"
+        role="switch"
         {...getStyles('input')}
       />
 
       <Box
-        component="label"
-        htmlFor={uuid}
+        aria-hidden="true"
```

Comment:
i would argue that the aria-hidden is not needed, as this will not be read if empty, but this will prevent any text from be read if onLabel  or offLabel are used

Reply:
after re reading the accessibility documentation found at https://www.w3.org/WAI/ARIA/apg/patterns/switch/examples/switch-checkbox/ i retract my question